packages/rxtranscribe/demo/runDemo.js

```diff
@@ -95,29 +105,70 @@ const schema = {
   }
 };
 
+const outputWriter = filePath => message$ => {
+  const rawJsonStr$ = message$.pipe(
+    scan(([index], message) => [index + 1, message], [-1, null]),
+    map(([index, message]) => ({index, ...message})),
+    map(message => JSON.stringify(message)),
+    share()
+  );
+  // add commas to all but the last json object
+  const allButLastJsonStr$ = rawJsonStr$.pipe(
+    // skipLast(1),
+    map(json => `${json},`)
+  );
+  // const lastJsonStr$ = rawJsonStr$.pipe(takeLast(1));
+  // const json$ = merge(allButLastJsonStr$, lastJsonStr$);
+  const json$ = allButLastJsonStr$;
+  const fileContentObservables = [
+    of('['), // starting character
+    json$, // JSON objects
+    of(']'), // ending character
+  ];
+  const outputWriter$ = concat(...fileContentObservables).pipe(
+    trace('outputToWrite'),
+    map(str => Buffer.from(str)),
+    trace('bufferToWrite'),
+    writeFile({filePath}),
+    trace('writeResult')
+  );
+  return outputWriter$;
+};
```

Comment:
My goal is to write the output to a file.  But there is a weird issue happening:
* The `message$` observable is just a series of messages (the raw STT output from each service). 
* For some reason, the last JSON message does not ever get written to the file.  
* Also, the end character (`']'`) never gets written to the file
* The file does include all of the JSON strings with commas, except the last JSON message is omitted
* When I run the node.js script (`yarn demo:run`) using DeepGram, it says "Done in X seconds" indicating that the script completed.  
* BUT the console doesn't log the "DONE" message that is supposed to log when the transcription$ observable (see line 167 below).  
* So it appears that the script is running through to completion but the observable never completes!  

I'm not sure how this is possible!  It seems like for the script to finish, either the transcription$ observable should complete or it should surface an error.  But neither of those things is happening!  I assume it's because the `json$` observable never completes so the last two items never get written to the file.  But if the observable isn't completing, how can the script be finishing?

Reply:
I guess what should happen is there should be a `stop$` trigger that causes the `transcription$` observable to complete...  But it surprises me that that this script can finish successfully...